src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1195,8 +1201,28 @@ internal SingleLookupResult CheckViability(Symbol symbol, int arity, LookupOptio
             {
                 return LookupResult.Good(symbol);
             }
-        }
 
+            bool IsBadIvtSpecification()
+            {
+                if ((unwrappedSymbol.DeclaredAccessibility == Accessibility.Internal || unwrappedSymbol.DeclaredAccessibility == Accessibility.ProtectedAndInternal 
+                    || unwrappedSymbol.DeclaredAccessibility == Accessibility.ProtectedOrInternal)
+                // Ensures that during binding we don't ask for public key which results in attribute binding and stack overflow.
+                // If looking up attributes, don't ask for public key.
```

Comment:
I think these comment lines are out of order (it reads better if the lower one is at the top).